analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Binary.cs

```diff
@@ -61,9 +61,30 @@ protected override ProgramState PreProcess(ProgramState state, IBinaryOperationW
     {
         BinaryOperatorKind.Add => NumberConstraint.From(left.Min + right.Min, left.Max + right.Max),
         BinaryOperatorKind.Subtract => NumberConstraint.From(left.Min - right.Max, left.Max - right.Min),
+        BinaryOperatorKind.Multiply => CalculateMultiply(left, right),
         _ => null
     };
 
+    private static NumberConstraint CalculateMultiply(NumberConstraint left, NumberConstraint right)
+    {
+        var products = new[] { left.Min * right.Min, left.Min * right.Max, left.Max * right.Min, left.Max * right.Max };
+        var min = products.Min();
```

Comment:
We don't need to precompute this in case the next `if` returns true.. so could be
```
var min = && || && || && ||
   ? null
   : products.Min();
```

Reply:
Is it possible to flip the condition the other way around? To end with `products.Min()` on the `true` branch?